// microsoft-cognitiveservices-speech-sdk/distrib/lib/src/common.speech/ServiceMessages/PhraseDetection/Enrichment.js
"use strict";
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapitalizationMode = exports.ProfanityHandlingMode = void 0;
/**
 * Profanity handling options.
 */
var ProfanityHandlingMode;
(function (ProfanityHandlingMode) {
    /**
     * This is the default behavior. The Microsoft Speech Service masks profanity with asterisks.
     */
    ProfanityHandlingMode["Masked"] = "Masked";
    /**
     * The Microsoft Speech Service removes profanity from all results.
     */
    ProfanityHandlingMode["Removed"] = "Removed";
    /**
     * The Microsoft Speech Service recognizes and returns profanity in all results.
     */
    ProfanityHandlingMode["Raw"] = "Raw";
    /**
     * The Microsoft Speech Service will surround profane words by XML tags &lt;profanity&gt; ... &lt;/profanity&gt;
     */
    ProfanityHandlingMode["Tagged"] = "Tagged";
    /**
     * The Microsoft Speech Service will add profanity label to the Words
     */
    ProfanityHandlingMode["Labeled"] = "Labeled";
})(ProfanityHandlingMode = exports.ProfanityHandlingMode || (exports.ProfanityHandlingMode = {}));
/**
 * The capitalization mode
 */
var CapitalizationMode;
(function (CapitalizationMode) {
    /**
     * Enable capitalization
     */
    CapitalizationMode["Enabled"] = "Enabled";
    /**
     * Disable capitalization
     */
    CapitalizationMode["Disabled"] = "Disabled";
})(CapitalizationMode = exports.CapitalizationMode || (exports.CapitalizationMode = {}));
