// microsoft-cognitiveservices-speech-sdk/distrib/lib/src/common.speech/ServiceMessages/PronunciationScore/PronunciationScoreContext.js
"use strict";
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
Object.defineProperty(exports, "__esModule", { value: true });
exports.DimensionKind = exports.GranularityKind = exports.GradingSystemKind = void 0;
/**
 * The grading system for the score
 */
var GradingSystemKind;
(function (GradingSystemKind) {
    /**
     * Five-point grading system
     */
    GradingSystemKind["FivePoint"] = "FivePoint";
})(GradingSystemKind = exports.GradingSystemKind || (exports.GradingSystemKind = {}));
/**
 * The granularity for score coverage
 */
var GranularityKind;
(function (GranularityKind) {
    /**
     * Phoneme granularity
     */
    GranularityKind["Phoneme"] = "Phoneme";
})(GranularityKind = exports.GranularityKind || (exports.GranularityKind = {}));
/**
 * The dimension of the score
 */
var DimensionKind;
(function (DimensionKind) {
    /**
     * Basic dimension
     */
    DimensionKind["Basic"] = "Basic";
})(DimensionKind = exports.DimensionKind || (exports.DimensionKind = {}));
