// microsoft-cognitiveservices-speech-sdk/distrib/lib/src/common.speech/ServiceMessages/PronunciationScore/PronunciationAssessmentOptions.js
"use strict";
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhonemeAlphabetKind = exports.DimensionKind = exports.GranularityKind = exports.GradingSystemKind = void 0;
/**
 * The enum of grading system for the score
 */
var GradingSystemKind;
(function (GradingSystemKind) {
    GradingSystemKind["FivePoint"] = "FivePoint";
    GradingSystemKind["HundredMark"] = "HundredMark";
})(GradingSystemKind = exports.GradingSystemKind || (exports.GradingSystemKind = {}));
/**
 * The enum of granularity for score coverage
 */
var GranularityKind;
(function (GranularityKind) {
    GranularityKind["Phoneme"] = "Phoneme";
    GranularityKind["Word"] = "Word";
    GranularityKind["FullText"] = "FullText";
})(GranularityKind = exports.GranularityKind || (exports.GranularityKind = {}));
/**
 * The enum of dimension of the score
 */
var DimensionKind;
(function (DimensionKind) {
    DimensionKind["Basic"] = "Basic";
    DimensionKind["Comprehensive"] = "Comprehensive";
})(DimensionKind = exports.DimensionKind || (exports.DimensionKind = {}));
/**
 * The kind of phoneme alphabet
 */
var PhonemeAlphabetKind;
(function (PhonemeAlphabetKind) {
    PhonemeAlphabetKind["SAPI"] = "SAPI";
    PhonemeAlphabetKind["IPA"] = "IPA";
})(PhonemeAlphabetKind = exports.PhonemeAlphabetKind || (exports.PhonemeAlphabetKind = {}));
